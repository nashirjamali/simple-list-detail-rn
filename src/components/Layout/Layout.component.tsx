import React, {FC} from 'react';
import {SafeAreaView} from 'react-native-safe-area-context';
import {StyleSheet, type StyleProp, type ViewStyle} from 'react-native';

import type {LayoutProps} from '../../@types';
import styles from './Layout.styles';

/**
 * A layout component that provides a safe area view for its children.
 *
 * @component
 * @param {LayoutProps} props - The properties for the Layout component.
 * @param {React.ReactNode} props.children - The content to be displayed within the layout.
 * @param {StyleProp<ViewStyle>} [props.style] - Additional styles to apply to the layout container.
 * @returns {JSX.Element} The rendered layout component.
 */
const Layout: FC<LayoutProps> = ({
  children,
  style,
}: LayoutProps): JSX.Element => {
  const combinedStyle = StyleSheet.flatten([styles.container, style]);

  return <SafeAreaView style={combinedStyle}>{children}</SafeAreaView>;
};

export default Layout;
