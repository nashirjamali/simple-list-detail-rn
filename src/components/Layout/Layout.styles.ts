import {StyleSheet} from 'react-native';

import type {LayoutStyles} from '../../@types';

const styles: LayoutStyles = StyleSheet.create({
  container: {
    padding: 10,
  },
});

export default styles;
