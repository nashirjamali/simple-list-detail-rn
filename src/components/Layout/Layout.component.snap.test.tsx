import React from 'react';
import {Text} from 'react-native';
import {render} from '@testing-library/react-native';

import Layout from './Layout.component';

describe('Layout Component', () => {
  it('renders correctly with children', () => {
    const {toJSON} = render(
      <Layout>
        <Text>Test Child</Text>
      </Layout>,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly with additional styles', () => {
    const {toJSON} = render(
      <Layout style={{backgroundColor: 'red'}}>
        <Text>Test Child with Style</Text>
      </Layout>,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
