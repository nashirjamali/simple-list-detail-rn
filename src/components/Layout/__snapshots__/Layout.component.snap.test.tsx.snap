// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Component renders correctly with additional styles 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    {
      "backgroundColor": "red",
      "padding": 10,
    }
  }
>
  <Text>
    Test Child with Style
  </Text>
</RNCSafeAreaView>
`;

exports[`Layout Component renders correctly with children 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    {
      "backgroundColor": "#F5F9F8",
      "padding": 10,
    }
  }
>
  <Text>
    Test Child
  </Text>
</RNCSafeAreaView>
`;
